parser/parser.go

```diff
@@ -37,77 +37,193 @@ func (e *ParseError) Error() string {
 	return fmt.Sprintf("%s: %s", e.Pos, e.Err)
 }
 
-type File struct {
-	Name     string
-	Defs     []Definition
-	Comments []*CommentGroup
+type ParseTree struct {
```

Comment:
The distinction between ParseTree and SyntaxTree is not clear to me.  Leaving this as File and File.Name would be much easier to review, and then you can use a separate patch to do renames.  The gorename command line tool is useful for doing programmatic renames, and you can include the gorename command in the commit message.